container/build_tar.py

```diff
@@ -178,6 +192,32 @@ def add_link(self, symlink, destination):
     symlink = os.path.normpath(symlink)
     self.tarfile.add_file(symlink, tarfile.SYMTYPE, link=destination)
 
+  @contextmanager
+  def write_temp_file(self, data, suffix='tar', mode='wb'):
+    (_, tmpfile) = tempfile.mkstemp(suffix=suffix)
+    try:
+      with open(tmpfile, mode='wb') as f:
+        f.write(data)
+      yield tmpfile
+    finally:
+      os.remove(tmpfile)
+
+  def add_pkg_metadata(self, metadata_tar, deb):
+    try:
+      with tarfile.open(metadata_tar) as tar:
+        # Metadata is expected to be in a file.
+        control_file = tar.extractfile(TarFile.PKG_METADATA_FILE)
+        metadata = ''.join(control_file.readlines())
+        destination_file = os.path.join(TarFile.DPKG_STATUS_DIR,
+                                        TarFile.parse_pkg_name(metadata, deb))
+        with self.write_temp_file(data=metadata) as metadata_file:
+          self.add_file(metadata_file, destination_file)
+    except (KeyError, TypeError) as e:
+      raise self.DebError(deb + ' contains invalid Metadata!')
+    except Exception as e:
+      raise self.DebError('Unknown Exception {0}. Please report an issue at'
+                          ' github.com:bazelbuild/rules_docker.'.format(e))
```

Comment:
Should this be github.com/ instead of github.com:

Reply:
done